php-checks/src/main/java/org/sonar/php/checks/HttpOnlyCheck.java

```diff
@@ -56,7 +56,25 @@
   @Override
   public void visitFunctionCall(FunctionCallTree tree) {
     if (isSetCookie(tree) && httpOnlySetToFalse(tree)) {
-      context().newIssue(this, tree.callee(), MESSAGE).secondary(tree.arguments().get(HTTP_ONLY_PARAMETER_INDEX), null);
+    
+      if (tree.arguments().size() > HTTP_ONLY_PARAMETER_INDEX) {
```

Comment:
I would rework the flow to avoid the duplication of this condition:
```
    if (isSetCookie(tree)) {
      if (tree.arguments().size() > HTTP_ONLY_PARAMETER_INDEX) {
        ExpressionTree httpOnlyArgument = tree.arguments().get(HTTP_ONLY_PARAMETER_INDEX);
        if (httpOnlyArgument.is(Kind.BOOLEAN_LITERAL) && CheckUtils.isFalseValue(httpOnlyArgument)) {
          context().newIssue(this, tree.callee(), MESSAGE).secondary(tree.arguments().get(HTTP_ONLY_PARAMETER_INDEX), null);  
        }
      } else if (tree.arguments().size() != 3) {
        context().newIssue(this, tree.callee(), MESSAGE);
      }
    }
```